plugin.rb

```diff
@@ -76,60 +101,28 @@ class Engine < ::Rails::Engine
   on(:user_added_to_group) do |user, group|
     name = DiscourseAutomation::Triggerable::USER_ADDED_TO_GROUP
 
-    DiscourseAutomation::Trigger.where(name: name).find_each do |trigger|
-      if trigger.metadata['group_ids'].include?(group.id)
-        trigger.run!(
+    DiscourseAutomation::Automation.where(trigger: name).find_each do |automation|
+      joined_group = automation.trigger_field('joined_group')
+      if joined_group['group_id'] == group.id
+        automation.trigger!(
           'kind' => DiscourseAutomation::Triggerable::USER_ADDED_TO_GROUP,
-          'user' => user,
+          'users' => [user],
           'group' => group
         )
       end
     end
   end
 
   on(:post_created) do |post|
-    if post.user_id != Discourse.system_user.id
-      name = DiscourseAutomation::Triggerable::POST_CREATED_EDITED
-
-      DiscourseAutomation::Trigger
-        .where(name: name)
-        .find_each do |trigger|
-          if trigger.metadata['category_id']
-            category_id = post.topic&.category&.parent_category&.id || post.topic&.category&.id
-            next if trigger.metadata['category_id'] != category_id
-          end
-
-          trigger.run!(
-            'kind' => DiscourseAutomation::Triggerable::POST_CREATED_EDITED,
-            'action' => :create,
-            'post' => post
-          )
-        end
-    end
+    handle_post_created_edited(post, :create)
   end
 
   on(:post_edited) do |post|
-    if post.user_id != Discourse.system_user.id
-      name = DiscourseAutomation::Triggerable::POST_CREATED_EDITED
-
-      DiscourseAutomation::Trigger
-        .where(name: name)
-        .find_each do |trigger|
-          if trigger.metadata['category_id']
-            category_id = post.topic&.category&.parent_category&.id || post.topic&.category&.id
-            next if trigger.metadata['category_id'] != category_id
-          end
-
-          trigger.run!(
-            'kind' => DiscourseAutomation::Triggerable::POST_CREATED_EDITED,
-            'action' => :edit,
-            'post' => post
-          )
-        end
-    end
+    handle_post_created_edited(post, :create)
```

Comment:
Shouldn't this be?

```suggestion
    handle_post_created_edited(post, :edited)
```

Reply:
yes indeed, it's weird that it's not triggering a failing spec, will make sur it does

Reply:
ok fixed and improved